### PR TITLE
[Feature] add deactivate devices guide

### PR DESCRIPTION
**Pull request by ceciliastevens**
**Description of Change**

This adds a user guide to clarify how to use the `code42 devices list` and `code42 devices bulk deactivate` commands in combination. We've had several internal questions about how to do this and what the capabilities are over the past few days, so I wanted to get something added to the documentation.

I've added the guide, but I'm not sure if there needs to be a changelog update for documentation changes. Should I put a line there?

Please also feel free to make any edits or changes you feel are necessary or recommended to the guide, it's very much a first draft at the moment.

### CODE REVIEW

**Comment by alanag13** [comment]
Thanks. You don not need to include a changelog update for documentation changes, that's just for things that affect usage / functionality. I'm adding @annie-payseur to the review to give feedback on the docs.

File: docs/userguides/deactivatedevices.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Clean up your environment by deactivating devices
+
+Your Code42 environment may contain many old devices that are no longer active computers, and that have not connected to Code42 in quite some time. In order to clean up your environment, you can use the CLI to deactivate these devices in bulk.
```
Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:3]: Lines are bit long - obviously only appears so in editors, but it doesn't hurt to put newlines in.

Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:3]: The comma is not needed after `computers`

File: docs/userguides/deactivatedevices.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Generating a list of devices
+
+You can generate a list of devices using `code42 devices list`. By default it will display the list of devices at the command line, but you can also output it in a number of file formats. For example, to generate a CSV of devices in your environment, use this command:
```
Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:7]: Missing comma after `By default`

File: docs/userguides/deactivatedevices.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+```
+code42 devices list -f CSV > output.csv
+```
+
+### Filtering the list
+
+You can filter or edit the list of devices in your spreadsheet or text editor of choice, but the CLI has some parameters built in that can help you to filter the list of devices to just the ones you want to deactivate. To see a full list of available parameters, run `code42 devices list -h`.
+
+Here are some useful paramaters you may wish to leverage when curating a list of devices to deactivate:
```
Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:23]: Spelling of `parameters`

File: docs/userguides/deactivatedevices.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+### Generating the list and deactivating in a single command
+
+You can also pipe the output of `code42 devices list` directly to `code42 devices bulk deactivate`. When using a pipe, make sure to use `-` as the input argument for `code42 devices bulk deactivate` to indicate that it should read from standard input.
+
+Here is an example:
+
+```
+code42 devices list --last-connected-before 365d --exclude-most-recently-connected 1 -f CSV | code42 devices bulk deactivate - -f CSV > deactivation_results.csv
```
Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:62]: For the longer commands, trailing `\` (line continuations) would be nice

Comment (unparalleled-js) [inline docs/userguides/deactivatedevices.md:62]: Extra `-` before `-f CSV` ?

**Review by unparalleled-js — COMMENTED** [review]
Thanks for this. Some pretty small feedback

File: docs/userguides/deactivatedevices.md (hunk 4 above)
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md:62]: It's not extra, that's necessary to tell the CLI to read from stdin instead of from a file.

Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: updated

File: docs/userguides/deactivatedevices.md (hunk 3 above)
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: updated

File: docs/userguides/deactivatedevices.md (hunk 2 above)
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: updated

File: docs/userguides/deactivatedevices.md (hunk 1 above)
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: updated

Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: I've added some line breaks. It's pretty arbitrary at this point so we may want to move them around later.

**Review by annie-payseur — CHANGES_REQUESTED** [review]
A few editorial suggestions.

File: docs/userguides/deactivatedevices.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+# Clean up your environment by deactivating devices
+
+Your Code42 environment may contain many old devices that are no
+longer active computers and that have not connected to Code42 in
+quite some time. In order to clean up your environment, you can
+use the CLI to deactivate these devices in bulk.
+
+## Generating a list of devices
+
```
[the remark this answers is not in the document]
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md:9]: Updated

File: docs/userguides/deactivatedevices.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+parameters:
+
+* `--change-device-name` - prepends `deactivated_<current_date>` to the beginning of the device name, allowing you to have a record of which devices were deactivated by the CLI and when.
+* `--purge-date yyyy-MM-dd` - allows you to change the date on which the deactivated devices' archives will be purged from cold storage.
+
+To see a full list of available options, run `code42 devices bulk deactivate -h`.
+
+The `code42 devices bulk deactivate` command will output the input
+you gave it, plus a column indicating the success or failure of
+the deactivation. To change the format of this output, use the
```
[the remark this answers is not in the document]
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: I've rephrased this. The `code42 devices bulk deactivate` command can be used standalone, but it does need to be provided an input spreadsheet. The output is the guid plus an indication of whether or not the deactivation was successful, which hopefully the new wording communicates better.

File: docs/userguides/deactivatedevices.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+Here are some useful parameters you may wish to leverage when
+curating a list of devices to deactivate:
+
+* `--last-connected-before DATE|TIMESTAMP|SHORT_TIME` - allows you to only see devices that have not connected since a particular date. You can also use a timestamp or short time format, for example `30d`.
+* `--exclude-most-recently-connected INTEGER` - allows you to exclude the most recently connected device (per user) from the results. This allows you to ensure that every user is left with at least N device(s), regardless of how recently they have connected.
+* `--created-before DATE|TIMESTAMP|SHORT_TIME` - allows you to only see devices created before a particular date.
+
+## Deactivating devices
+
+Once you have a list of devices that you wish to remove, you can
```
[the remark this answers is not in the document]
Comment (ceciliastevens) [reply docs/userguides/deactivatedevices.md]: updated!

**Review by annie-payseur — APPROVED** [review]

**Pull request merged**